corehq/apps/users/migrations/0046_add_default_mobile_worker_role.py

```diff
@@ -12,16 +12,14 @@
 def _add_default_mobile_worker_role(apps, schema_editor):
     default_mobile_worker_roles = UserRole.objects.filter(is_commcare_user_default=True)
     for domain in Domain.get_all():
-        dmw_roles_in_domain = default_mobile_worker_roles.filter(domain=domain)
-        if len(dmw_roles_in_domain) == 0:
+        has_dmw_role = default_mobile_worker_roles.filter(domain=domain).exists()
```

Comment:
oh, just noticed this.  `domain` here is a domain object - you should probably be using `Domain.get_all_names()`.  Weirdly enough, this will usually work as written because of this
https://github.com/dimagi/commcare-hq/blob/1bd5f684cbf6d0fc08624dcc395b20a8e4429604/corehq/apps/domain/models.py#L744-L749
But it won't find default roles on any projects that have specified a human-readable name.

Reply:
No worries, it's an easy enough fix - just pushed a PR to make sure the change gets in before the overnight prod deploy (since prod is probably the only env where this matters)
https://github.com/dimagi/commcare-hq/pull/31783

Reply:
And the domain string vs object thing is a recurring source of issues.  The docstring for `Domain` hints at some past frustrations :laughing: :
https://github.com/dimagi/commcare-hq/blob/8073bebb636d5e5dba2ba025fecb2fcc9cd1c40f/corehq/apps/domain/models.py#L290-L297